`CODE_TO_TOKEN_ERROR` inserts full token in SESSION_ID column

### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

core

### Describe the bug

I noticed that I'm getting Stacktraces when a client (Rancher) tries to receive a token. I suspected to be inserted in the column so I temporarily altered the table `EVENT_ID`.`SESSION_ID` to be of type text instead and I noticed that the full encoded token is inserted in the session id:

```
Oct 24 15:23:42 X000VV30 kc.sh[1734733]: 2024-10-24 15:23:42,192 WARN  [org.keycloak.events] (executor-thread-14) type="CODE_TO_TOKEN_ERROR", realmId="107f816d-85b9-4408-90d5-4e17371f6bd9", realmName="sso", clientId="rancher", userId="null", sessionId="eyJleHAiOjE3Mjk3MDc3OTQsImlhdCI6MTcyOTcwNzUwNywiYXV0aF90aW1lIjoxNzI5NjY0NTk0LCJqdGkiOiIwMTUyM2MwOS1mZmMyLTRhNzctYTQ2ZS1hMDE3ODM1NjhhYzMiLCJpc3MiOiJodHRwczovL2F1dGgtdGVzdC5pbnRlcm4ucmVkYWN0ZWQuY29tL3JlYWxtcy9zc28iLCJhdWQiOlsicmFuY2hlciIsImFjY291bnQiXSwic3ViIjoiZWRmMDhkNjAtNzM3Yy00OWZiLTg3MmEtNGE1YjM4MWQxMDE0IiwidHlwIjoiQmVhcmVyIiwiYXpwIjoicmFuY2hlciIsInNpZCI6ImExYjJlMzBjLTQyODEtNGM0Ny1iNzg3LWQwNWEwOTEyN2IxZSIsImFjciI6IjEiLCJyZWFsbV9hY2Nlc3MiOnsicm9sZXMiOlsib2ZmbGluZV9hY2Nlc3MiLCJkZWZhdWx0LXJvbGVzLXNzbyIsInVtYV9hdXRob3JpemF0aW9uIl19LCJyZXNvdXJjZV9hY2Nlc3MiOnsiYWNjb3VudCI6eyJyb2xlcyI6WyJtYW5hZ2UtYWNjb3VudCIsIm1hbmFnZS1hY2NvdW50LWxpbmtzIiwidmlldy1wcm9maWxlIl19fSwic2NvcGUiOiJvcGVuaWQgcHJvZmlsZSBlbWFpbCIsImVtYWlsX3ZlcmlmaWVkIjp0cnVlLCJuYW1lIjoiUmVkYWN0ZWQgUmVkYWN0ZWQiLCJwcmVmZXJyZWRfdXNlcm5hbWUiOiJyZWRhY3RlZCIsImdpdmVuX25hbWUiOiJyZWRhY3RlZCIsImZhbWlseV9uYW1lIjoiUmVkYWN0ZWQiLCJlbWFpbCI6InJlZGFjdGVkQGV4YW1wbGUuY29tIn0=", ipAddress="10.45.194.242", error="invalid_code", grant_type="authorization_code", code_id="eyJleHAiOjE3Mjk3MDc3OTQsImlhdCI6MTcyOTcwNzUwNywiYXV0aF90aW1lIjoxNzI5NjY0NTk0LCJqdGkiOiIwMTUyM2MwOS1mZmMyLTRhNzctYTQ2ZS1hMDE3ODM1NjhhYzMiLCJpc3MiOiJodHRwczovL2F1dGgtdGVzdC5pbnRlcm4ucmVkYWN0ZWQuY29tL3JlYWxtcy9zc28iLCJhdWQiOlsicmFuY2hlciIsImFjY291bnQiXSwic3ViIjoiZWRmMDhkNjAtNzM3Yy00OWZiLTg3MmEtNGE1YjM4MWQxMDE0IiwidHlwIjoiQmVhcmVyIiwiYXpwIjoicmFuY2hlciIsInNpZCI6ImExYjJlMzBjLTQyODEtNGM0Ny1iNzg3LWQwNWEwOTEyN2IxZSIsImFjciI6IjEiLCJyZWFsbV9hY2Nlc3MiOnsicm9sZXMiOlsib2ZmbGluZV9hY2Nlc3MiLCJkZWZhdWx0LXJvbGVzLXNzbyIsInVtYV9hdXRob3JpemF0aW9uIl19LCJyZXNvdXJjZV9hY2Nlc3MiOnsiYWNjb3VudCI6eyJyb2xlcyI6WyJtYW5hZ2UtYWNjb3VudCIsIm1hbmFnZS1hY2NvdW50LWxpbmtzIiwidmlldy1wcm9maWxlIl19fSwic2NvcGUiOiJvcGVuaWQgcHJvZmlsZSBlbWFpbCIsImVtYWlsX3ZlcmlmaWVkIjp0cnVlLCJuYW1lIjoiUmVkYWN0ZWQgUmVkYWN0ZWQiLCJwcmVmZXJyZWRfdXNlcm5hbWUiOiJyZWRhY3RlZCIsImdpdmVuX25hbWUiOiJyZWRhY3RlZCIsImZhbWlseV9uYW1lIjoiUmVkYWN0ZWQiLCJlbWFpbCI6InJlZGFjdGVkQGV4YW1wbGUuY29tIn0=", client_auth_method="client-secret"

[root@X000VV30 root]# mysql -D keycloak
Reading table information for completion of table and column names
You can turn off this feature to get a quicker startup with -A

Welcome to the MariaDB monitor.  Commands end with ; or \g.
Your MariaDB connection id is 518
Server version: 10.5.22-MariaDB MariaDB Server

Copyright (c) 2000, 2018, Oracle, MariaDB Corporation Ab and others.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

MariaDB [keycloak]> SELECT SESSION_ID FROM EVENT_ENTITY LIMIT 1;
+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| SESSION_ID                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          |
+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| eyJleHAiOjE3Mjk3MDc3OTQsImlhdCI6MTcyOTcwNzUwNywiYXV0aF90aW1lIjoxNzI5NjY0NTk0LCJqdGkiOiIwMTUyM2MwOS1mZmMyLTRhNzctYTQ2ZS1hMDE3ODM1NjhhYzMiLCJpc3MiOiJodHRwczovL2F1dGgtdGVzdC5pbnRlcm4ucmVkYWN0ZWQuY29tL3JlYWxtcy9zc28iLCJhdWQiOlsicmFuY2hlciIsImFjY291bnQiXSwic3ViIjoiZWRmMDhkNjAtNzM3Yy00OWZiLTg3MmEtNGE1YjM4MWQxMDE0IiwidHlwIjoiQmVhcmVyIiwiYXpwIjoicmFuY2hlciIsInNpZCI6ImExYjJlMzBjLTQyODEtNGM0Ny1iNzg3LWQwNWEwOTEyN2IxZSIsImFjciI6IjEiLCJyZWFsbV9hY2Nlc3MiOnsicm9sZXMiOlsib2ZmbGluZV9hY2Nlc3MiLCJkZWZhdWx0LXJvbGVzLXNzbyIsInVtYV9hdXRob3JpemF0aW9uIl19LCJyZXNvdXJjZV9hY2Nlc3MiOnsiYWNjb3VudCI6eyJyb2xlcyI6WyJtYW5hZ2UtYWNjb3VudCIsIm1hbmFnZS1hY2NvdW50LWxpbmtzIiwidmlldy1wcm9maWxlIl19fSwic2NvcGUiOiJvcGVuaWQgcHJvZmlsZSBlbWFpbCIsImVtYWlsX3ZlcmlmaWVkIjp0cnVlLCJuYW1lIjoiUmVkYWN0ZWQgUmVkYWN0ZWQiLCJwcmVmZXJyZWRfdXNlcm5hbWUiOiJyZWRhY3RlZCIsImdpdmVuX25hbWUiOiJyZWRhY3RlZCIsImZhbWlseV9uYW1lIjoiUmVkYWN0ZWQiLCJlbWFpbCI6InJlZGFjdGVkQGV4YW1wbGUuY29tIn0= |
+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.001 sec)

```

### Version

26.0.2

### Regression

- [ ] The issue is a regression

### Expected behavior

The authentication to fail with a proper or an empty SESSION_ID instead of the token/code being inserted in the EVENT_ENTITY table

### Actual behavior

The encoded token is inserted and causes a Stacktrace to be printed for every attempt.

### How to Reproduce?

I'm not sure what Rancher(v2.9.1) does to provoke this error, we followed their steps [here](https://ranchermanager.docs.rancher.com/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/authentication-config/configure-keycloak-oidc) for the integration and followed every step

### Anything else?

Logs from the service: 
```
Oct 24 14:41:38 X000VV30 kc.sh[2476]: 2024-10-24 14:41:38,808 WARN  [org.hibernate.engine.jdbc.spi.SqlExceptionHelper] (executor-thread-2846) SQL Error: 1406, SQLState: 22001
Oct 24 14:41:38 X000VV30 kc.sh[2476]: 2024-10-24 14:41:38,808 ERROR [org.hibernate.engine.jdbc.spi.SqlExceptionHelper] (executor-thread-2846) (conn=494) Data too long for column 'SESSION_ID' at row 1
Oct 24 14:41:38 X000VV30 kc.sh[2476]: 2024-10-24 14:41:38,808 WARN  [com.arjuna.ats.arjuna] (executor-thread-2846) ARJUNA012125: TwoPhaseCoordinator.beforeCompletion - failed for SynchronizationImple< 0:ffff0a2dc5ad:811b:670d7124:107495, org.hibernate.resource.transaction.backend.jta.internal.synchronization.RegisteredSynchronization@19e778f >: org.hibernate.exception.DataException: could not execute batch [(conn=494) Data too long for column 'SESSION_ID' at row 1] [insert into EVENT_ENTITY (CLIENT_ID,DETAILS_JSON,DETAILS_JSON_LONG_VALUE,ERROR,IP_ADDRESS,REALM_ID,SESSION_ID,EVENT_TIME,TYPE,USER_ID,ID) values (?,?,?,?,?,?,?,?,?,?,?)]
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.hibernate.exception.internal.SQLStateConversionDelegate.convert(SQLStateConversionDelegate.java:103)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.hibernate.exception.internal.StandardSQLExceptionConverter.convert(StandardSQLExceptionConverter.java:58)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:108)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.hibernate.engine.jdbc.batch.internal.BatchImpl.lambda$performExecution$2(BatchImpl.java:294)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.hibernate.engine.jdbc.mutation.internal.PreparedStatementGroupSingleTable.forEachStatement(PreparedStatementGroupSingleTable.java:67)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.hibernate.engine.jdbc.batch.internal.BatchImpl.performExecution(BatchImpl.java:264)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.hibernate.engine.jdbc.batch.internal.BatchImpl.execute(BatchImpl.java:242)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.hibernate.engine.jdbc.internal.JdbcCoordinatorImpl.executeBatch(JdbcCoordinatorImpl.java:188)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.hibernate.engine.spi.ActionQueue.executeActions(ActionQueue.java:674)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.hibernate.engine.spi.ActionQueue.executeActions(ActionQueue.java:511)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.hibernate.event.internal.AbstractFlushingEventListener.performExecutions(AbstractFlushingEventListener.java:414)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.hibernate.event.internal.DefaultFlushEventListener.onFlush(DefaultFlushEventListener.java:41)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.hibernate.event.service.internal.EventListenerGroupImpl.fireEventOnEachListener(EventListenerGroupImpl.java:127)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.hibernate.internal.SessionImpl.doFlush(SessionImpl.java:1429)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.hibernate.internal.SessionImpl.managedFlush(SessionImpl.java:491)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.hibernate.internal.SessionImpl.flushBeforeTransactionCompletion(SessionImpl.java:2354)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.hibernate.internal.SessionImpl.beforeTransactionCompletion(SessionImpl.java:1978)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.hibernate.engine.jdbc.internal.JdbcCoordinatorImpl.beforeTransactionCompletion(JdbcCoordinatorImpl.java:439)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.hibernate.resource.transaction.backend.jta.internal.JtaTransactionCoordinatorImpl.beforeCompletion(JtaTransactionCoordinatorImpl.java:336)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.hibernate.resource.transaction.backend.jta.internal.synchronization.SynchronizationCallbackCoordinatorNonTrackingImpl.beforeCompletion(SynchronizationCallbackCoordinatorNonTrackingImpl.java:47)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.hibernate.resource.transaction.backend.jta.internal.synchronization.RegisteredSynchronization.beforeCompletion(RegisteredSynchronization.java:37)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at com.arjuna.ats.internal.jta.resources.arjunacore.SynchronizationImple.beforeCompletion(SynchronizationImple.java:52)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at com.arjuna.ats.arjuna.coordinator.TwoPhaseCoordinator.beforeCompletion(TwoPhaseCoordinator.java:348)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at com.arjuna.ats.arjuna.coordinator.TwoPhaseCoordinator.end(TwoPhaseCoordinator.java:66)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at com.arjuna.ats.arjuna.AtomicAction.commit(AtomicAction.java:135)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at com.arjuna.ats.internal.jta.transaction.arjunacore.TransactionImple.commitAndDisassociate(TransactionImple.java:1290)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at com.arjuna.ats.internal.jta.transaction.arjunacore.BaseTransaction.commit(BaseTransaction.java:104)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at io.quarkus.narayana.jta.runtime.NotifyingTransactionManager.commit(NotifyingTransactionManager.java:70)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.keycloak.transaction.JtaTransactionWrapper.commit(JtaTransactionWrapper.java:90)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.keycloak.services.DefaultKeycloakTransactionManager.commit(DefaultKeycloakTransactionManager.java:136)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.keycloak.services.DefaultKeycloakSession.closeTransactionManager(DefaultKeycloakSession.java:414)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.keycloak.services.DefaultKeycloakSession.close(DefaultKeycloakSession.java:379)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.keycloak.models.utils.KeycloakModelUtils.runJobInTransactionWithResult(KeycloakModelUtils.java:399)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.keycloak.models.utils.KeycloakModelUtils.runJobInTransaction(KeycloakModelUtils.java:275)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.keycloak.events.EventBuilder.send(EventBuilder.java:242)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.keycloak.events.EventBuilder.error(EventBuilder.java:229)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.keycloak.protocol.oidc.grants.AuthorizationCodeGrantType.process(AuthorizationCodeGrantType.java:81)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.keycloak.protocol.oidc.endpoints.TokenEndpoint.processGrantRequest(TokenEndpoint.java:151)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.keycloak.protocol.oidc.endpoints.TokenEndpoint$quarkusrestinvoker$processGrantRequest_6408e15340992839b66447750c221d9aaa837bd7.invoke(Unknown Source)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.jboss.resteasy.reactive.server.handlers.InvocationHandler.handle(InvocationHandler.java:29)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at io.quarkus.resteasy.reactive.server.runtime.QuarkusResteasyReactiveRequestContext.invokeHandler(QuarkusResteasyReactiveRequestContext.java:141)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:147)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:635)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2516)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2495)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1521)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:11)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:11)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at java.base/java.lang.Thread.run(Thread.java:840)
Oct 24 14:41:38 X000VV30 kc.sh[2476]: Caused by: java.sql.BatchUpdateException: (conn=494) Data too long for column 'SESSION_ID' at row 1
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.mariadb.jdbc.export.ExceptionFactory.createBatchUpdate(ExceptionFactory.java:221)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.mariadb.jdbc.client.impl.StandardClient.executePipeline(StandardClient.java:1000)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.mariadb.jdbc.ClientPreparedStatement.executeBatchPipeline(ClientPreparedStatement.java:165)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.mariadb.jdbc.ClientPreparedStatement.executeInternalPreparedBatch(ClientPreparedStatement.java:146)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.mariadb.jdbc.BasePreparedStatement.executeBatch(BasePreparedStatement.java:1623)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at io.agroal.pool.wrapper.StatementWrapper.executeBatch(StatementWrapper.java:340)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.hibernate.engine.jdbc.batch.internal.BatchImpl.lambda$performExecution$2(BatchImpl.java:279)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   ... 46 more
Oct 24 14:41:38 X000VV30 kc.sh[2476]: Caused by: java.sql.SQLSyntaxErrorException: (conn=494) Data too long for column 'SESSION_ID' at row 1
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.mariadb.jdbc.export.ExceptionFactory.createException(ExceptionFactory.java:289)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.mariadb.jdbc.export.ExceptionFactory.create(ExceptionFactory.java:378)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.mariadb.jdbc.message.ClientMessage.readPacket(ClientMessage.java:189)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.mariadb.jdbc.client.impl.StandardClient.readPacket(StandardClient.java:1235)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.mariadb.jdbc.client.impl.StandardClient.readResults(StandardClient.java:1174)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.mariadb.jdbc.client.impl.StandardClient.readResponse(StandardClient.java:1093)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.mariadb.jdbc.client.impl.StandardClient.executePipeline(StandardClient.java:933)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   ... 51 more
Oct 24 14:41:38 X000VV30 kc.sh[2476]: 2024-10-24 14:41:38,808 ERROR [org.keycloak.services.error.KeycloakErrorHandler] (executor-thread-2846) Uncaught server error: org.keycloak.models.ModelException: Database operation failed
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.keycloak.connections.jpa.PersistenceExceptionConverter.convert(PersistenceExceptionConverter.java:109)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.keycloak.connections.jpa.JpaExceptionConverter.convert(JpaExceptionConverter.java:31)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.keycloak.transaction.JtaTransactionWrapper.lambda$handleException$0(JtaTransactionWrapper.java:65)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at java.base/java.util.HashMap$ValueSpliterator.forEachRemaining(HashMap.java:1779)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.keycloak.transaction.JtaTransactionWrapper.handleException(JtaTransactionWrapper.java:67)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.keycloak.transaction.JtaTransactionWrapper.commit(JtaTransactionWrapper.java:92)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.keycloak.services.DefaultKeycloakTransactionManager.commit(DefaultKeycloakTransactionManager.java:136)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.keycloak.services.DefaultKeycloakSession.closeTransactionManager(DefaultKeycloakSession.java:414)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.keycloak.services.DefaultKeycloakSession.close(DefaultKeycloakSession.java:379)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.keycloak.models.utils.KeycloakModelUtils.runJobInTransactionWithResult(KeycloakModelUtils.java:399)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.keycloak.models.utils.KeycloakModelUtils.runJobInTransaction(KeycloakModelUtils.java:275)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.keycloak.events.EventBuilder.send(EventBuilder.java:242)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.keycloak.events.EventBuilder.error(EventBuilder.java:229)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.keycloak.protocol.oidc.grants.AuthorizationCodeGrantType.process(AuthorizationCodeGrantType.java:81)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.keycloak.protocol.oidc.endpoints.TokenEndpoint.processGrantRequest(TokenEndpoint.java:151)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.keycloak.protocol.oidc.endpoints.TokenEndpoint$quarkusrestinvoker$processGrantRequest_6408e15340992839b66447750c221d9aaa837bd7.invoke(Unknown Source)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.jboss.resteasy.reactive.server.handlers.InvocationHandler.handle(InvocationHandler.java:29)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at io.quarkus.resteasy.reactive.server.runtime.QuarkusResteasyReactiveRequestContext.invokeHandler(QuarkusResteasyReactiveRequestContext.java:141)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:147)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:635)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2516)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2495)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1521)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:11)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:11)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at java.base/java.lang.Thread.run(Thread.java:840)
Oct 24 14:41:38 X000VV30 kc.sh[2476]: Caused by: java.sql.BatchUpdateException: (conn=494) Data too long for column 'SESSION_ID' at row 1
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.mariadb.jdbc.export.ExceptionFactory.createBatchUpdate(ExceptionFactory.java:221)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.mariadb.jdbc.client.impl.StandardClient.executePipeline(StandardClient.java:1000)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.mariadb.jdbc.ClientPreparedStatement.executeBatchPipeline(ClientPreparedStatement.java:165)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.mariadb.jdbc.ClientPreparedStatement.executeInternalPreparedBatch(ClientPreparedStatement.java:146)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.mariadb.jdbc.BasePreparedStatement.executeBatch(BasePreparedStatement.java:1623)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at io.agroal.pool.wrapper.StatementWrapper.executeBatch(StatementWrapper.java:340)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.hibernate.engine.jdbc.batch.internal.BatchImpl.lambda$performExecution$2(BatchImpl.java:279)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.hibernate.engine.jdbc.mutation.internal.PreparedStatementGroupSingleTable.forEachStatement(PreparedStatementGroupSingleTable.java:67)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.hibernate.engine.jdbc.batch.internal.BatchImpl.performExecution(BatchImpl.java:264)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.hibernate.engine.jdbc.batch.internal.BatchImpl.execute(BatchImpl.java:242)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.hibernate.engine.jdbc.internal.JdbcCoordinatorImpl.executeBatch(JdbcCoordinatorImpl.java:188)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.hibernate.engine.spi.ActionQueue.executeActions(ActionQueue.java:674)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.hibernate.engine.spi.ActionQueue.executeActions(ActionQueue.java:511)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.hibernate.event.internal.AbstractFlushingEventListener.performExecutions(AbstractFlushingEventListener.java:414)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.hibernate.event.internal.DefaultFlushEventListener.onFlush(DefaultFlushEventListener.java:41)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.hibernate.event.service.internal.EventListenerGroupImpl.fireEventOnEachListener(EventListenerGroupImpl.java:127)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.hibernate.internal.SessionImpl.doFlush(SessionImpl.java:1429)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.hibernate.internal.SessionImpl.managedFlush(SessionImpl.java:491)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.hibernate.internal.SessionImpl.flushBeforeTransactionCompletion(SessionImpl.java:2354)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.hibernate.internal.SessionImpl.beforeTransactionCompletion(SessionImpl.java:1978)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.hibernate.engine.jdbc.internal.JdbcCoordinatorImpl.beforeTransactionCompletion(JdbcCoordinatorImpl.java:439)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.hibernate.resource.transaction.backend.jta.internal.JtaTransactionCoordinatorImpl.beforeCompletion(JtaTransactionCoordinatorImpl.java:336)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.hibernate.resource.transaction.backend.jta.internal.synchronization.SynchronizationCallbackCoordinatorNonTrackingImpl.beforeCompletion(SynchronizationCallbackCoordinatorNonTrackingImpl.java:47)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.hibernate.resource.transaction.backend.jta.internal.synchronization.RegisteredSynchronization.beforeCompletion(RegisteredSynchronization.java:37)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at com.arjuna.ats.internal.jta.resources.arjunacore.SynchronizationImple.beforeCompletion(SynchronizationImple.java:52)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at com.arjuna.ats.arjuna.coordinator.TwoPhaseCoordinator.beforeCompletion(TwoPhaseCoordinator.java:348)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at com.arjuna.ats.arjuna.coordinator.TwoPhaseCoordinator.end(TwoPhaseCoordinator.java:66)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at com.arjuna.ats.arjuna.AtomicAction.commit(AtomicAction.java:135)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at com.arjuna.ats.internal.jta.transaction.arjunacore.TransactionImple.commitAndDisassociate(TransactionImple.java:1290)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at com.arjuna.ats.internal.jta.transaction.arjunacore.BaseTransaction.commit(BaseTransaction.java:104)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at io.quarkus.narayana.jta.runtime.NotifyingTransactionManager.commit(NotifyingTransactionManager.java:70)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.keycloak.transaction.JtaTransactionWrapper.commit(JtaTransactionWrapper.java:90)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   ... 21 more
Oct 24 14:41:38 X000VV30 kc.sh[2476]: Caused by: java.sql.SQLSyntaxErrorException: (conn=494) Data too long for column 'SESSION_ID' at row 1
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.mariadb.jdbc.export.ExceptionFactory.createException(ExceptionFactory.java:289)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.mariadb.jdbc.export.ExceptionFactory.create(ExceptionFactory.java:378)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.mariadb.jdbc.message.ClientMessage.readPacket(ClientMessage.java:189)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.mariadb.jdbc.client.impl.StandardClient.readPacket(StandardClient.java:1235)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.mariadb.jdbc.client.impl.StandardClient.readResults(StandardClient.java:1174)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.mariadb.jdbc.client.impl.StandardClient.readResponse(StandardClient.java:1093)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   at org.mariadb.jdbc.client.impl.StandardClient.executePipeline(StandardClient.java:933)
Oct 24 14:41:38 X000VV30 kc.sh[2476]:   ... 51 more
Oct 24 14:41:38 X000VV30 kc.sh[2476]: 2024-10-24 14:41:38,842 WARN  [org.keycloak.events] (executor-thread-2846) type="REFRESH_TOKEN_ERROR", realmId="107f816d-85b9-4408-90d5-4e17371f6bd9", realmName="sso", clientId="rancher", userId="null", ipAddress="10.45.194.242", error="invalid_token", reason="Token is not active", grant_type="refresh_token", client_auth_method="client-secret"
Oct 24 14:41:38 X000VV30 kc.sh[2476]: 2024-10-24 14:41:38,844 WARN  [org.keycloak.events] (executor-thread-2846) type="REFRESH_TOKEN_ERROR", realmId="107f816d-85b9-4408-90d5-4e17371f6bd9", realmName="sso", clientId="rancher", userId="null", ipAddress="10.45.194.242", error="invalid_token", reason="Token is not active", grant_type="refresh_token", client_auth_method="client-secret"
Oct 24 14:42:27 X000VV30 kc.sh[2476]: 2024-10-24 14:42:27,632 WARN  [org.mariadb.jdbc.message.server.ErrorPacket] (executor-thread-2846) Error: 1406-22001: Data too long for column 'SESSION_ID' at row 1
Oct 24 14:42:27 X000VV30 kc.sh[2476]: 2024-10-24 14:42:27,632 INFO  [org.hibernate.orm.jdbc.batch] (executor-thread-2846) HHH100503: On release of batch it still contained JDBC statements
Oct 24 14:42:27 X000VV30 kc.sh[2476]: 2024-10-24 14:42:27,632 ERROR [org.hibernate.orm.jdbc.batch] (executor-thread-2846) HHH100501: Exception executing batch [java.sql.BatchUpdateException: (conn=494) Data too long for column 'SESSION_ID' at row 1], SQL: insert into EVENT_ENTITY (CLIENT_ID,DETAILS_JSON,DETAILS_JSON_LONG_VALUE,ERROR,IP_ADDRESS,REALM_ID,SESSION_ID,EVENT_TIME,TYPE,USER_ID,ID) values (?,?,?,?,?,?,?,?,?,?,?)
```

--- a/server-spi-private/src/main/java/org/keycloak/events/Event.java
+++ b/server-spi-private/src/main/java/org/keycloak/events/Event.java
@@ -107,7 +107,7 @@ public class Event {
     }
 
     public void setSessionId(String sessionId) {
-        this.sessionId = sessionId;
+        this.sessionId = maxLength(sessionId, 255);
     }
 
     public String getIpAddress() {

--- a/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/events/EventStoreProviderTest.java
+++ b/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/events/EventStoreProviderTest.java
@@ -246,6 +246,9 @@ public class EventStoreProviderTest extends AbstractEventsTest {
         testing().onEvent(create(System.currentTimeMillis() - 30000, EventType.LOGIN, StringUtils.repeat(realmId, 100), "clientId", "userId", "127.0.0.1", "error"));
         testing().onEvent(create(System.currentTimeMillis() - 30000, EventType.LOGIN, realmId, "clientId", StringUtils.repeat("userId", 100), "127.0.0.1", "error"));
 
+        EventRepresentation event = create(System.currentTimeMillis() - 30000, EventType.LOGIN, realmId, "clientId", "userId", "127.0.0.1", "error");
+        event.setSessionId(StringUtils.repeat("sessionId", 100));
+        testing().onEvent(event);
     }
 
     @Test

